src/main/java/io/odpf/firehose/sink/log/LogSinkforJson.java

```diff
@@ -0,0 +1,50 @@
+package io.odpf.firehose.sink.log;
+
+import io.odpf.firehose.error.ErrorInfo;
+import io.odpf.firehose.error.ErrorType;
+import io.odpf.firehose.message.Message;
+import io.odpf.firehose.metrics.Instrumentation;
+import io.odpf.firehose.sink.AbstractSink;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class LogSinkforJson extends AbstractSink {
```

Comment:
Hmm. instead of creating a separate class, can we have a different flow in the same class? Only thing will change is parsing..so we can abstract out parsing logic..
```
interface MessageParser { 
  Object parse(Message m); 
}
class ProtoMessageParser implements MessageParser{
   
DynamicMessage parse(Message m) {}

}
class JsonMessageParser implements MessageParser {
JsonObject parse(Message m) {}
}
```
and then 
```
LogSink(MessageParser, Instrumentation);
 protected List<Message> execute() throws Exception {
 for (Message m : messageList) {
   instrumentation.logInfo(messageParser.parse(m).toString());
   }
 }
```

Reply:
For logSink this makes sense, but the motivation for creating separate class LogSinkForJson is when we have other sinks like BigquerySink, HTTP sink, for each input format apart from just parsing there would be more complex logic involved , So having separate class for each input format ie json or protobuf

Reply:
We can still have separate class for sink and have abstraction for parser.

Reply:
implemented the suggested changes, updated the PR